ReactSkia/sdk/ThreadSafeCache.h

```diff
@@ -19,39 +21,79 @@ using namespace facebook;
 template <typename KeyT, typename ValueT>
 class ThreadSafeCache {
  private:
-  static  int overallCurrentSize_;
-  static  int overallMaxSize_;
+  double overallCurrentSize_;
   std::mutex cacheLock_;
-  int currentSize_;
-  int maxSize_;
   KeyT key_;
   better::map <KeyT,ValueT> cacheMap_;
+  Timer * timer_{nullptr};
+  double scheduleTimeExpiry_;
  public:
   bool isAvailableInCache(const KeyT &key){
     auto it = cacheMap_.find(key);
     if(it!= cacheMap_.end())
       return true;
     return false;
-  };
+  }
+
   better::optional<ValueT> getCacheData(const KeyT &key) {
     std::scoped_lock lock(cacheLock_);
     auto it = cacheMap_.find(key);
     if(it == cacheMap_.end()) {
        return{};
     }
     return it->second;
-  };
-  ThreadSafeCache(int currentSize, int maxsize):currentSize_(currentSize),maxSize_(maxsize){
-      
-  };
-  void setCache(const KeyT &key ,  const ValueT &value) {
+  }
+
+  void expiryTimerCallback() {
+    auto it =cacheMap_.begin();
+    double currentTime = Timer::getCurrentTimeMSecs();
+    std::chrono::duration<double, std::milli> milliseconds = Timer::getFutureTime().time_since_epoch();
+    double scheduleTimeExpiry = milliseconds.count();
+    while(it != cacheMap_.end()) {
+      if(it->second->cacheExpiryTime <= currentTime) {
```

Comment:
we are checking cacheExpiryTime from the ValueT .
Since this is a template class , ValueT can be of any type and it may or may not have cacheExpiryTime data member right ?